### DISCUSSION TITLE
Declarative Entity Definition

### DISCUSSION THREAD

**Discussion by philhawthorn** [noam-honig/radweb#312] [Q&A] [answered]
Hi, 

I'm wondering about declarative entity definitions. 

For example, if I want to allow my users to extend the underlying database with their own entities (albeit in a controlled manner) from _within_ a remult react app. I've experimented with `describeClass` to read json config files with success, but was wondering if there are any red flags that immediately jump out with that approach? 

Thanks!

**Comment by noam-honig** [comment, marked as answer]
I use that approach a lot in testing - and I'm happy with it there.

**Reply by philhawthorn** [reply]
Perfect, thanks for the speedy reply. I'll have a look at some of the tests for inspiration 👍
